server/database/repositories/TeamRepository.ts

```diff
@@ -1,11 +1,66 @@
 import { TeamWithoutMS } from '../../../shared/types/Tournament';
-import TeamModel from '../models/TeamModel';
+import { UserWithoutMS } from '../../../shared/types/User';
+import AppError from '../../utils/appError';
+import TeamModel, { TeamDocument, UserDocument } from '../models/TeamModel';
+
+// change _id to id and delete __v
+// This can't be done on mongoose layer
+interface UserDb extends Omit<UserWithoutMS, 'id'> {
+  _id: string,
+}
+function toUserDb(usr: UserWithoutMS): UserDb {
+  const usrDb : UserDb = { _id: usr.id, ...usr };
+  return usrDb;
+}
```

Comment:
```suggestion
function toUserDb(usr: UserWithoutMS): UserDb {
  return { _id: usr.id, ...usr };
}
```

When you add returns type, you can return object like this. This also applies to the rest

Reply:
corrected